posts/2022-01-14-construct-vendia-univseral-application-using-domain-driven-design.md

```diff
@@ -0,0 +1,177 @@
+# Construct Vendia Universal Application using Domain Driven Design
+
+Vendia Share is a data platform that you can use to quickly build real-time data sharing applications, which we call Universal Applications (or "Unis" for short). You can build both internal and external applications on Vendia Share. Internal applications include applications that you need for enterprise data management - transactional, analytical, and master data. External applications include applications for data exchange across trust boundaries between partners or a consortium. 
+
+Data models power Vendia Share.  The schema-first approach makes it easy to deploy and operate a Uni. In fact, deploy and operate are almost indistinguishable steps after you have a ready-to-deploy schema. Vendia Share provides you with a GraphQL API that you can use to read and write data into Vendia Share and it also generates an immutable log of all transactions. Because a data model is a critical piece needed to create a Uni, designing the right data model for your application is an essential step to use Vendia Share. 
+
+In this blog, you will first go through a primer on Domain Driven Design(DDD), and then you will learn how to apply DDD when constructing a Uni. As a developer, you don’t have to learn a different design approach to build strategic applications on the Vendia Share platform. DDD is a popular and proven approach to designing and modeling systems that are strategic for your business. You can use the same concepts to design and model Uni(s). DDD is not a prerequisite to building Uni but it does help to create better software design. If DDD is new to you, you may benefit from reading Eric Evans' [book](https://www.domainlanguage.com/wp-content/uploads/2016/05/DDD_Reference_2015-03.pdf). 
+
+
+## Primer on Domain Driven Design(DDD)
+
+DDD is a popular software design approach for architecting and designing software using a top-down approach. DDD is a valuable exercise because it creates a common language that binds business and software. Requirements are dynamic in nature, they are good in defining activities, behaviors, rules, and actors. But, the data models built on requirements are fragile. DDD domain data models are static in nature and bring stability to data models, DDD is good in defining information that includes - its structure and relationships. DDD domain models delimit(set boundaries) and refine(make it precise) requirements. 
```

Comment:
```suggestion
DDD is a popular top-down approach to software architecture and design. DDD is a valuable exercise because it creates a common language that binds business and technology. Requirements are dynamic in nature, they are good in defining activities, behaviors, rules, and actors. But, the data models built on requirements are fragile. DDD domain models are static in nature and bring stability to data models, DDD is good in defining information that includes - its structure and relationships. DDD domain models delimit (i.e. set boundaries) and refine requirements (i.e. make them precise). 
```